tethys_apps/templatetags/site_settings.py

```diff
@@ -14,15 +14,25 @@
 @register.filter
 @stringfilter
 def load_custom_css(var):
+    """Load Custom Styles defined in Tethys Portal -> Site Settings
+
+    Args:
+        var: a filename of CSS to load or CSS text to embed into the page
+
+    Returns: a string of HTML that either embeds CSS text or points to a file
+
+    """
     if var.startswith("/"):
         var = var.lstrip("/")
 
-    if (Path(settings.STATIC_ROOT) / var).is_file() or static_finder.find(var):
-        return f'<link href="/static/{var}" rel="stylesheet" />'
+    try:
+        if (Path(settings.STATIC_ROOT) / var).is_file() or static_finder.find(var):
+            return f'<link href="/static/{var}" rel="stylesheet" />'
 
-    else:
         for path in settings.STATICFILES_DIRS:
             if (Path(path) / var).is_file():
                 return f'<link href="/static/{var}" rel="stylesheet" />'
+    except OSError:
+        pass
```

Comment:
Is the assumption here that if the sting is too long, an OSError will be thrown and therefore it's not a file path, so it must be a css string? What about paths that really are too long? It seems like we should make our checking logic more robust and validate that the string is css if the OSError is raised and if not, log a warning.

Reply:
That's what I was going for. I also thought about specifically targeting Errno 36, but any OSError should steer it towards a CSS string.

I will add something here to check if `var` has any curly braces or semicolons or commas, and then log an error if not. Something like this plus setting up logger and testing it:
```suggestion
        common_css_chars = "{};,"
        if not any(c in var for c in common_css_chars):
            # This appears to be a filename and not a CSS string
            logger.warning("Could not load file '%s' for custom styles. %s", var, str(e))
            return ""
```

Reply:
That seems like a reasonable approach for detecting the CSS. Whatever feedback we can provide to the user/administrator would be great.

Reply:
- Added filename detection code and log warning to check all input that isn't a file. This should help show problems with a missing file or misspelled filename, not just the rare thing that could trigger an OSError.
- load_custom_css() gets called a lot, so I added a quick return if `var` is empty. That also prevents false positives with the log warning.
- Updated test_get_css_is_code() to test with CSS code instead of "test.css", since load_custom_css() now returns an empty string if it detects a filename that can't be found.